XenAdmin/Dialogs/HealthCheck/HealthCheckOverviewDialog.cs

```diff
@@ -279,15 +279,26 @@ public void UpdateAnalysisResult(HealthCheckSettings healthCheckSettings)
                 previousUploadPanel.Visible = false;
             }
         }
-
+        
         public void UpdateButtonsVisibility(Pool pool)
         {
+            bool canAlterHealthCheck = Core.HealthCheck.PassedRbacChecks(pool.Connection);

```

Comment:
In order to write a better message (see my comment in Messages.resx) we should extend the PassedRbacChecks method to return the allowed roles for the operation. To do this replace in the latter method `CanPerform(RbacMethodList, IXenConnection)` with `public static bool CanPerform(RbacMethodList, IXenConnection, out List<Role>)`.